apps/aecore/lib/aecore/structures/signed_tx.ex

```diff
@@ -3,33 +3,54 @@ defmodule Aecore.Structures.SignedTx do
   Aecore structure of a signed transaction.
   """
 
-  alias Aecore.Keys.Worker, as: Keys
   alias Aecore.Structures.TxData
   alias Aecore.Structures.SignedTx
 
+  @typedoc "Structure of the TxData"
+  @type tx() :: %TxData{}
+
+  @typedoc "Structure of the SignedTx module"
   @type t :: %SignedTx{
-    data: TxData.t(),
+    data: tx(),
     signature: binary()
   }
 
   @doc """
-    Definition of Aecore SignedTx structure
+  Definition of Aecore SignedTx structure
 
   ## Parameters
-     - data: Aecore %TxData{} structure
-     - signature: Signed %TxData{} with the private key of the sender
+     * data: Aecore %TxData{} structure
+     * signature: Signed %TxData{} with the private key of the sender
   """
   defstruct [:data, :signature]
   use ExConstructor
 
-  @spec is_coinbase?(SignedTx.t()) :: boolean()
-  def is_coinbase?(tx) do
-    tx.data.from_acc == nil && tx.signature == nil
+  @spec is_coinbase(signed_tx()) :: boolean()
+  def is_coinbase(%{data: %{from_acc: key}, signature: signature}) do
+    key == nil && signature == nil
+  end
+
+  @spec is_valid(signed_tx()) :: boolean()
+  def is_valid(%{data: data, signature: signature}) do
+    data.value >= 0 && data.fee >= 0
+      Aewallet.Signing.verify(:erlang.term_to_binary(data), signature, data.from_acc)
```

Comment:
is there an `&&` missing? otherwise, how does the verification work?

Reply:
Yes, my fault. I saw it later, it's fixed